oracle/internal/dkg/dkg_executor.go

```diff
@@ -254,45 +285,52 @@ func (e *Executor) executeR2(dkg *coordinator.DKG, validatorIdx uint16) bool {
 			return false
 		}
 
-		e.artifacts.r2 = &Round2Result{
-			pkgs:   r2Packages,
-			secret: NewSecret(r2SecretPtr),
-		}
-	}
+		// Convert R2 packages to bytes
+		var r2packagesBatched []byte
+		tmpBuf := make([]byte, 2)
+		for toIdentificator, r2pkg := range r2Packages {
+			// Serialize validator idx
+			toValidatorIdx := helpers.FrostToValidatorIdx(toIdentificator)
+			binary.BigEndian.PutUint16(tmpBuf, toValidatorIdx)
+			r2packagesBatched = append(r2packagesBatched, tmpBuf...)
+
+			// Encrypt r2pkg
+			r2PublicKeyX25519, ok := r2PublicKeysX25519[toValidatorIdx]
```

Comment:
Call it before 
`toValidatorIdx := helpers.FrostToValidatorIdx(toIdentificator)`